src/main/java/br/com/sbk/sbking/gui/constants/FrameConstants.java

```diff
@@ -6,59 +6,42 @@
 
 import br.com.sbk.sbking.core.Direction;
 
-// (TODO) refactor this class to reduce instruction duplication. 
 public final class FrameConstants {
 	public static final java.awt.Color TABLE_COLOR = new java.awt.Color(0, 100, 0); // Tablecloth green
-	public static int TABLE_WIDTH = 1024;
-	public static int TABLE_HEIGHT = 768;
+	public static int TABLE_WIDTH;
+	public static int TABLE_HEIGHT;
 
-	public static int HALF_WIDTH = TABLE_WIDTH / 2;
-	public static int HALF_HEIGHT = TABLE_HEIGHT / 2;
-	private static int FIFTH_WIDTH = TABLE_WIDTH / 5;
-	private static int FIFTH_HEIGHT = TABLE_HEIGHT / 5;
-
-	private static int NORTH_X_CENTER = HALF_WIDTH;
-	private static int NORTH_Y_CENTER = FIFTH_HEIGHT;
-
-	private static int SOUTH_X_CENTER = HALF_WIDTH;
-	private static int SOUTH_Y_CENTER = FIFTH_HEIGHT * 4;
-
-	private static int EAST_X_CENTER = FIFTH_WIDTH * 4;
-	private static int EAST_Y_CENTER = HALF_HEIGHT;
-
-	private static int WEST_X_CENTER = FIFTH_WIDTH;
-	private static int WEST_Y_CENTER = HALF_HEIGHT;
+	public static int HALF_WIDTH;
+	public static int HALF_HEIGHT;
 
 	@SuppressWarnings("serial")
```

Comment:
This line should go down to the HashMap construction

Reply:
Better yet, this could be initialized here and only updated on the computeContants(..). This way, older references cannot be saved throughout the program.

So:
```
public static Map<Direction, Point> pointOfDirection = new HashMap<Direction, Point>();
```
and:
```
pointOfDirection.clear();
		pointOfDirection.put(Direction.NORTH, new Point(NORTH_X_CENTER, NORTH_Y_CENTER));
		pointOfDirection.put(Direction.EAST, new Point(EAST_X_CENTER, EAST_Y_CENTER));
		pointOfDirection.put(Direction.SOUTH, new Point(SOUTH_X_CENTER, SOUTH_Y_CENTER));
		pointOfDirection.put(Direction.WEST, new Point(WEST_X_CENTER, WEST_Y_CENTER));
```

Reply:
Ah, your explanation makes sense.

> This way, older references cannot be saved throughout the program.
in our particular case, if someone cached the return of pointOfDirection this issue still holds. Solving this particular issue is arguably outside of the scope of this PR, and this is not an issue today in the code base. I'll add a comment above the variable to keep track of this.